boards/shields/apollo4p_evb_disp_shield_rev2/apollo4p_evb_disp_shield_rev2.overlay

```diff
@@ -0,0 +1,22 @@
+/*
+ *  Copyright (c) 2023 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	aliases {
+		accel0 = &adxl362;
+	};
+};
+
+&ambiq_spi1 {
```

Comment:
IS ambiq_spi1 a well-known connector?

"Hardware shield-to-board compatibility depends on the use of well-known connectors used on popular boards (such as Arduino and 96boards)." https://docs.zephyrproject.org/latest/hardware/porting/shields.html

Reply:
The connector defined is ambiq_header inside apollo4p_evb_connector.dtsi.
This is also where ambiq_spi1 is defined.

Since we define out own ambiq_conncetor, we use the 'ambiq_' precursor to the spi.
Similar to arduino.

Reply:
If not ambiq_spi, what should it be?

Reply:
Is this connector on your board (hardware) standardized and documented somewhere, or is it just a random type? Are there other boards that use this type?

@erwango Maybe we should maintain a well-known connectors table in shield documentation.

Reply:
We defined ambiq-header to create our own header type.

I made the assumption following arduino-header  that I could define ambiq_spi similar to arduino_spi.

For me, I would rather do it however the community thinks we should.

If we shouldn't be defining our own header type, I'd rather fix that now.

Reply:
Fixed.